### PR TITLE
Create purge command

### PR DESCRIPTION
**Pull request by JasperGray**
Warning: if we're logging deletes for the purge command, then the delete messages will be rate-limited because the deletions are occurring too quickly.

### CODE REVIEW

File: Floofbot/Modules/Administration.cs [hunk 1]
```diff
@@ -146,6 +148,45 @@ public class Administration : ModuleBase<SocketCommandContext>
...
+            foreach (ISocketMessageChannel channel in Context.Guild.TextChannels)
+            {
+                var asyncMessageCollections = channel.GetMessagesAsync(MESSAGES_TO_SCAN_PER_CHANNEL_ON_PURGE);
+                await foreach(var messageCollection in asyncMessageCollections)
+                {
+                    foreach (var message in messageCollection)
+                    {
+                        if (message.Author.Id == badUser.Id)
+                        {
+                            await channel.DeleteMessageAsync(message);
```
Comment (iShauny) [inline Floofbot/Modules/Administration.cs:175]: could run this in a delayed async task to help with rate limits async task

File: Floofbot/Modules/Administration.cs [hunk 2]
```diff
@@ -146,6 +148,45 @@ public class Administration : ModuleBase<SocketCommandContext>
...
 
+        [Command("purge")]
+        [Alias("p")]
+        [Summary("Deletes recent messages from a given user for all channels on the server")]
+        [RequireContext(ContextType.Guild)]
+        [RequireUserPermission(GuildPermission.BanMembers)]
+        public async Task PurgeUserMessages(
+            [Summary("user")] string user)
+        {
+            IUser badUser = resolveUser(user);
```
Comment (iShauny) [inline Floofbot/Modules/Administration.cs:159]: will this command work after a user is banned?

File: Floofbot/Modules/Administration.cs (hunk 1 above)
Comment (JasperGray) [reply Floofbot/Modules/Administration.cs:175]: The part that needs to be delayed isn't exactly the deletion, but the logging of the messages

File: Floofbot/Modules/Administration.cs (hunk 2 above)
Comment (JasperGray) [reply Floofbot/Modules/Administration.cs:159]: All it has to do is resolve the parameter user, which is either the userid or the username#tag.
I'm thinking that if banning a user after they already left the server works, then this should work too.

File: Floofbot/Modules/Administration.cs (hunk 1 above)
Comment (iShauny) [reply Floofbot/Modules/Administration.cs:175]: delaying the deletion will delay the logging, though

File: Floofbot/Modules/Administration.cs (hunk 2 above)
Comment (iShauny) [reply Floofbot/Modules/Administration.cs:159]: it does not, though :( I think we wont be able to resolve this until we implement a user cache

**Pull request merged**